api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -157,6 +176,11 @@ message TlsParameters {
   //   rsa_pkcs1_sha1
   //   ecdsa_sha1
   repeated string signature_algorithms = 5;
+
+  // Compliance policy configures various aspects of the TLS based on the given policy.
+  // The policies are applied last during configuration and may override the other TLS
+  // parameters.
+  CompliancePolicy compliance_policy = 6;
```

Comment:
Just to make sure I fully understand this, an Envoy TLS-context is not expected to support more than one policy, correct?

Reply:
Correct; the boringssl API allows setting a single enum value.

Reply:
The reason I'm asking is because unfortunately we cannot really deprecate API features.
There are 2 things I'm trying to consider from the API point-of-view:
1) Whether this field should be repeated or not? currently it can support only a single entry, and later it can support more than one without adding new fields.
2) Whether it should use a new-message instead of an enum? This allows adding custom configuration for each entry.

I'm guessing the answer to 2 is no, and I cannot think of a reasonable future use-case where this will change.
But if there's a chance that Envoy may support more than one element, I think that the field should be set to repeated, and have a PGV requiring no more than one entry at the moment.

Reply:
Looking at the implementation in boringssl, it seems likely that setting a second policy may override some settings from the first policy, although there may be some sets of policies that don't overlap. But I guess there's not really much pain in making the API repeated instead of single. If we do that, need to take NONE out of the enum.

Reply:
>  If we do that, need to take NONE out of the enum.

That's a good point. Will it make explicitly state that if the field is not defined, it is considered as NONE (explaining the definition of None)?

Reply:
Looking at the doc string for `SSL_CTX_set_compliance_policy` it says that the function is only well-defined if it's the last thing called on context. Since calling it twice would require making one call undefined, we cannot really support that today, so it's effective a single policy today that can be enforced.

Reply:
In any case, I understand that we should keep the door open for multiple policies, updated the proto. PTAL.

Reply:
If it's going to be in a repeated field, then NONE should be removed from the enum

Reply:
Oh, I'm sorry, I was going to remove it, and forgot. Updated.